src/pymoca/backends/casadi/api.py

```diff
@@ -145,9 +145,20 @@ def _codegen_model(model_folder: str, f: ca.Function, library_name: str):
             os.remove(object_name)
     return library
 
-def _save_model(model_folder: str, model_name: str, model: Model,
-                compiler_options: Dict[str, str], cache=True, codegen=False):
+def save_model(model_folder: str, model_name: str, model: Model,
+               compiler_options: Dict[str, str]) -> None:
+    """
+    Saves a CasADi model to disk.
+
+    :param model_folder: Folder where the precompiled CasADi model will be stored.
+    :param model_name: Name of the model.
+    :param model: Model instance.
+    :param compiler_options: Dictionary of compiler options.
+    """
+
     # Compilation takes precedence over caching, and disables it
+    cache = compiler_options.get('cache', False)
+    codegen = compiler_options.get('codegen', False)
```

Comment:
Saving a model only makes sense if one of them is True. This was not necessary in `transfer_model` (as it would just _not_ save it), but should probably be part of the `save_model` if we make it public (and remove the default arguments).